de.variantsync.core/src/test/java/de/variantsync/core/JsonParserTestASTwithLineGrammar.java

```diff
@@ -0,0 +1,107 @@
+package de.variantsync.core;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import de.variantsync.core.ast.AST;
+import de.variantsync.core.ast.JsonParserASTwithLineGrammar;
+import de.variantsync.core.ast.LineGrammar;
+import org.junit.Before;
+import org.junit.Test;
+
+public class JsonParserTestASTwithLineGrammar {
+
+	AST<LineGrammar, String> exampleAst;
+	Path exmaplePath;
+
+	@Before
+	public void InitJsonTest() throws IOException {
+		// init
+		exampleAst = new AST<>(LineGrammar.Directory, "src");
+		AST<LineGrammar, String> mainJava = new AST<>(LineGrammar.TextFile, "Main.java");
+		exampleAst.addChild(mainJava);
+		mainJava.addChildren(
+				Arrays.asList(new AST<>(LineGrammar.Line, "public class Main {"), new AST<>(LineGrammar.Line, "    public static void main(String[] args)"),
+						new AST<>(LineGrammar.Line, "        System.out.println(\"Hello World\");"), new AST<>(LineGrammar.Line, "    }"),
+						new AST<>(LineGrammar.Line, "}")));
+
+		exmaplePath = Path.of("out.txt");
+
+	}
+
+	@Test
+	public void TestJsonParserAST() {
+
+		// export to json
+		String json = JsonParserASTwithLineGrammar.exportAST(exampleAst);
+
+		// import ast from json
+		AST<LineGrammar, String> ast = JsonParserASTwithLineGrammar.importAST(json);
```

Comment:
add `assertEquals(exampleAST, ast)`?
Of course that would require us to think about equality of ASTs...

Reply:
also use a more descriptive variable name such as `reimportedExampleAST`

Reply:
Writing a tree comparater is out of scope for now. Please add a todo here that when it is available in the future, we want to use it for comparing the trees here. Applies to similar comments for the other tests.